sharder/deterministic.go

```diff
@@ -217,20 +262,41 @@ func (d *DeterministicSharder) loadPeerList() error {
 			ipOrHost: peerURL.Hostname(),
 			port:     peerURL.Port(),
 		}
-		newPeers = append(newPeers, peerShard)
+		newPeers[ix] = peerShard
 	}
 
-	// the redis peer discovery already sorts its content. Does every backend?
-	// well, it's not too much work, let's sort it one more time.
+	// make sure the list is in a stable, comparable order
 	sort.Sort(SortableShardList(newPeers))
 
+	// PartitionCount controls the minimum number of partitions used to control node assignment
+	// when we use the "hash" strategy.
+	// Ideally, this value should be 3-5x the maximum expected number of nodes.
+	// If necessary, it will (effectively) be increased to equal the number of nodes.
+	// A larger value will achieve more even trace distribution, at the cost of a minor performance
+	// hit once per span (roughly 1 microsecond per 50 partitions). Since the distribution isn't
+	// that even anyway (yay randomness) 50 seems to be a good balance; it helps with small
+	// installations and doesn't hurt for big ones.
+	const partitionCount = 50
+	// now build the hash list;
+	// We make a list of hash value and an index to a peer.
+	hashes := make([]hashShard, 0)
+	hashesPerPeer := partitionCount/len(peerList) + 1
+	for ix := range newPeers {
+		hashes = append(hashes, newPeers[ix].GetHashesFor(ix, hashesPerPeer, peerSeed)...)
```

Comment:
Do we end up with `50 * len(newPeers)` unique hashes?  I admit it is my lack of knowledge about rendezvous hashing, but I am missing the reason why we need so many.  What would happen if `hashesPerPeer` was 1?